dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/MergeProcessor.java

```diff
@@ -0,0 +1,773 @@
+     */
+    private Map<OpenAPIProvider, Set<String>> pathNames = new HashMap<>();
+
+    /**
+     * The unique names and their values which are in use for models processed so far, grouped by the type of name
+     */
+    private Map<NameType, Map<String, Object>> namesInUse = new HashMap<>();
+
+    private List<String> mergeProblems = new ArrayList<>();
+
+    private void add(OpenAPIProvider document) {
+        providers.add(document);
+    }
+
+    private void process() {
+
+        List<InProgressModel> inProgressModels = new ArrayList<>();
+
+        // First loop which just does all the steps necessary to determine whether there are any fatal merge conflicts
+        // which result in removing some of the requested merge providers.
+        // This ensures that when we come to rename things to avoid non-fatal merge conflicts, we aren't trying to avoid
+        // clashing with models that have been removed.
+        for (OpenAPIProvider provider : providers) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                Tr.event(this, tc, "Starting path processing for " + provider);
+            }
+
+            // Make a deep copy, to avoid modifying the input model
+            OpenAPI model = (OpenAPI) ModelCopy.copy(provider.getModel());
+
+            RenameHolder renameHolder = new RenameHolder();
+
+            // Add context root to the front of paths if possible
+            prependPaths(model, provider.getApplicationPath(), renameHolder);
+
+            // Check for clashes
+            if (!findAndRecordPathClashes(model, provider)) {
+                InProgressModel inProgressModel = new InProgressModel();
+                inProgressModel.model = model;
+                inProgressModel.provider = provider;
+                inProgressModel.renameHolder = renameHolder;
+                inProgressModels.add(inProgressModel);
+
+                if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+                    Tr.event(this, tc, "No path clashes for " + provider + ", including in merge");
+                }
+            }
+        }
+
+        boolean securityIdentical = isSecurityIdentical(inProgressModels);
+        boolean infoIdentical = isInfoIdentical(inProgressModels);
+        boolean externalDocsIdentical = isExternalDocsIdentical(inProgressModels);
+        boolean serversIdentical = isServersIdentical(inProgressModels);
+
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(this, tc, "Security identical = " + securityIdentical);
+            Tr.event(this, tc, "Info identical = " + infoIdentical);
+            Tr.event(this, tc, "External docs identical = " + externalDocsIdentical);
+            Tr.event(this, tc, "Servers identical = " + serversIdentical);
+        }
+
+        // Second loop which moves and renames parts of the models where necessary to avoid clashes
```

Comment:
maybe some of the process method should be split out into separate methods?

Reply:
Which bits did you have in mind?

At the moment, the process method does very little other than call other private methods which do well defined tasks. As it is, I think it's fairly easy to see the complete set of steps taken to merge several models so I'm not sure that breaking it up any more would make it easier to read or understand.

There are some slightly verbose bits, maybe `InProgressModel` should have a constructor so that its creation doesn't take up so many lines. (It was added fairly late when I realised I needed two loops so that I could remove any models which couldn't be merged before I calculate whether the info, servers and security sections are identical across all models).

Reply:
I've shortened some of the verbose bits in the first loop.